data_structures/arrays/permutations.py

```diff
@@ -1,33 +1,32 @@
-def permute(nums: list[int]) -> list[list[int]]:
+def permute_iterative(nums: list[int]) -> list[list[int]]:
```

Comment:
```suggestion
def permute_recursive(nums: list[int]) -> list[list[int]]:
```
I'd argue that this function is a recursive implementation, not an iterative one, because we're constructing the permutations by repeatedly calling the function on smaller and smaller inputs

Reply:
Agreed. Didn't see the fact that it called itself